### ISSUE TITLE
Allow comments in multiline RC file commands

### ISSUE THREAD

**Issue by bovine3dom** [tridactyl/tridactyl#4555]
At the moment

```
js window.alert("hello \
" this is a comment
world")
```

doesn't work in our RC files.

We need to change the order in this function:

https://github.com/tridactyl/tridactyl/blob/e26865e4e4cbd000509bb72199ddb50bfc718a30/src/background/config_rc.ts#L61

so that we remove comments before we start joining lines together.

**Comment by mozbugbox** [comment]
Should also consider javascript single line comment: `//`.

**Comment by callmetushar123** [comment]
@bovine3dom Hello assign this also to me please

**Comment by bovine3dom** [comment]
In our RC files

```
js window.alert("hello \
" this is a comment
world")
```

should have the comment removed and then the escaped lines should be merged, like this:

```
js window.alert("hello \
" this is a comment
world")

remove comments -->

js window.alert("hello \
world")

merge lines -->

js window.alert("hello world")

```

What currently happens is

```
js window.alert("hello \
" this is a comment
world")

merge lines --->

js window.alert("hello " this is a comment
world")

remove comments -->

js window.alert("hello " this is a comment
world")
```


---

You can test it by installing the native messenger (`:native`), putting snippet in the original post in a file and then running `:source [~/path/to/file]` in Tridactyl :)

**Comment by bovine3dom** [comment]
Fixed in #4651
